hf_xet/src/logging.rs

```diff
@@ -103,45 +135,50 @@ fn init_global_logging(py: Python) -> Option<TelemetryTaskInfo> {
 
-    // Client-side telemetry, default is OFF
-    // To enable telemetry set env var HF_HUB_ENABLE_TELEMETRY
-    if env::var("HF_HUB_ENABLE_TELEMETRY").is_err_and(|e| e == env::VarError::NotPresent) {
-        let tr_sub = tracing_subscriber::registry().with(filter_layer);
+    // Do we use telemetry?
+    if env::var("HF_HUB_ENABLE_TELEMETRY").is_ok() {
+        match init_telemetry_logging(version_info) {
+            Ok(tl) => {
+                let telemetry_filter_layer = tl.with_filter(FilterFn::new(|meta| meta.target() == "client_telemetry"));
 
-        if use_json().unwrap_or(false) {
-            tr_sub.with(fmt_layer_base.json()).init();
-        } else {
-            tr_sub.with(fmt_layer_base.pretty()).init();
-        }
+                tracing_subscriber::registry()
+                    .with(filter_layer)
+                    .with(fmt_layer_base.json())
+                    .with(telemetry_filter_layer)
+                    .init();
 
-        None
-    } else {
-        let telemetry_buffer_layer = LogBufferLayer::new(py, TELEMETRY_PRE_ALLOC_BYTES);
-        let telemetry_task_info: TelemetryTaskInfo =
-            (telemetry_buffer_layer.buffer.clone(), telemetry_buffer_layer.stats.clone());
+                return;
+            },
 
-        let telemetry_filter_layer =
-            telemetry_buffer_layer.with_filter(FilterFn::new(|meta| meta.target() == "client_telemetry"));
+            Err(e) => {
+                eprintln!("Error initializing telemetry process : {e:?}. Reverting to logging to console.");
+            },
+        }
+    }
 
-        tracing_subscriber::registry()
-            .with(filter_layer)
-            .with(fmt_layer_base.json())
-            .with(telemetry_filter_layer)
-            .init();
+    // Now, just use basic console logging.
+    let tr_sub = tracing_subscriber::registry().with(filter_layer);
 
-        Some(telemetry_task_info)
+    if use_json().unwrap_or(false) {
+        tr_sub.with(fmt_layer_base.json()).init();
+    } else {
+        tr_sub.with(fmt_layer_base.pretty()).init();
     }
 }
 
-pub fn initialize_runtime_logging(py: Python, runtime: Arc<ThreadPool>) {
-    static GLOBAL_TELEMETRY_TASK_INFO: OnceLock<Option<TelemetryTaskInfo>> = OnceLock::new();
+static INITIALIZED_LOGGING_ID: AtomicU32 = AtomicU32::new(0);
 
-    // First get or init the global logging componenents.
-    let telemetry_task_info = GLOBAL_TELEMETRY_TASK_INFO.get_or_init(move || init_global_logging(py));
+pub fn check_logging_state(py: Python<'_>) {
+    let logger_pid = INITIALIZED_LOGGING_ID.load(Ordering::SeqCst);
 
-    // Spawn the telemetry logging.
-    if let Some(ref tti) = telemetry_task_info {
-        let telemetry_task = get_telemetry_task(tti.clone());
-        let _telemetry_task = runtime.spawn(telemetry_task);
+    let pid = std::process::id();
+
+    if logger_pid == 0 {
+        init_global_logging(py);
+        INITIALIZED_LOGGING_ID.store(pid, Ordering::SeqCst);
```

Comment:
`init_global_logging` needs to only be called once or else it will panic. As it is currently, we might have simultaneous threads call `check_logging_state` in parallel, which leads to a race-condition here (both threads detect that logger_pid == 0 before either one can initialize logging).

Instead of the AtomicU32, we probably want the ID to be a OnceCell to guarantee this is only run once?

Reply:
Not sure how often this race will occur though, but if something changes with pyo3 (especially with some non-GIL support), we might become susceptible to the race.